primary/tests/integration_tests_validator_api.rs

```diff
@@ -925,15 +928,18 @@ async fn test_get_collections_with_missing_certificates() {
         2,
         actual_result
             .iter()
-            .filter(|&r| matches!(r.retrieval_result, Some(types::RetrievalResult::Batch(_))))
+            .filter(|&r| matches!(
+                r.retrieval_result,
+                Some(types::RetrievalResult::TransactionList(_))
+            ))
             .count()
     );
 
     for result in actual_result {
         match result.retrieval_result.unwrap() {
-            RetrievalResult::Batch(batch) => {
-                let id: BatchDigest = batch.id.unwrap().into();
-                let result_batch: Batch = batch.transactions.unwrap().into();
+            RetrievalResult::TransactionList(transaction_list) => {
+                let id: BatchDigest = transaction_list.id.unwrap().into();
+                let result_batch: Batch = transaction_list.transactions.unwrap().into();
```

Comment:
This use of `BatchDigest` and `Batch` is the one that is illegitimate: the digest in here is a `CertificateDigest`, and the transaction list itself should have its own type.

Reply:
I actually threw away the `CertificateDigest` in the original implementation and only returned the `BatchDigest` and `Batch`. Explained further below.